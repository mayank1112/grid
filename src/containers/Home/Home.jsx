import { useEffect } from 'react';
import Button from '../../components/Button/Button';
import axios from 'axios';
import { useState } from 'react';

export const Home = () => {
    const [data, setData] = useState('');

    const generateGrid = async () => {
        const response = await axios.get('/mock.json');
        setData(response.data.value);
    };

    useEffect(() => {
        generateGrid();
    }, []);

    const onClick = () => {
        console.log('click');
    };

    return (
        <Button
            onClick={onClick}
        >{data}</Button>

    );
};