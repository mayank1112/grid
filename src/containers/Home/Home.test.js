import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import axios from 'axios';
import { act } from 'react-dom/cjs/react-dom-test-utils.development';
import React from 'react';
import toJson from 'enzyme-to-json';
import { Home } from './Home';

Enzyme.configure({ adapter: new Adapter() });

it('<Home /> component should call axios', () => {
    jest.spyOn(axios, 'get').mockImplementation(() => Promise.resolve({ data: { value: "2" } }));
    jest.spyOn(React, 'useState').mockImplementation(init => ["1", jest.fn()]);

    let wrapper;
    act(() => {
        wrapper = mount(<Home />);
        wrapper.update();
    });

    expect(axios.get).toHaveBeenCalledTimes(1);
});

it('<Home /> component should match snapshot', () => {
    jest.spyOn(axios, 'get').mockImplementation(() => Promise.resolve({ data: { value: "2" } }));
    jest.spyOn(React, 'useState').mockImplementation(init => ["1", jest.fn()]);

    const component = shallow(<Home />);
    expect(toJson(component)).toMatchSnapshot();
});
