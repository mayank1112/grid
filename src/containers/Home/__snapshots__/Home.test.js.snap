// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> component should match snapshot 1`] = `
<Box
  onClick={[Function]}
>
  1
</Box>
`;
